### PR TITLE
Add an optional 'fast' mode that uses more memory, but is fast

### PR DESCRIPTION
**Pull request by postwait**
for buckets with more than one count. O(log N) insert of new bucket,
but O(1) updated of existin buckets.

### CODE REVIEW

File: src/circllhist.c [hunk 1]
```diff
@@ -88,15 +88,23 @@ static const char __b64[] = {
...
+  uint32_t fast:1;
   struct {
     hist_bucket_t bucket;
     uint64_t count;
   } __attribute__((packed)) *bvs;
 };
-
+struct histogram_fast {
+  struct histogram internal;
+  uint16_t *faster[256];
```
Comment (rileyberton) [inline src/circllhist.c]: Is this meant to be an array of pointers or an array of values?

File: src/circllhist.c [hunk 2]
```diff
@@ -786,7 +816,15 @@ hist_num_buckets(const histogram_t *hist) {
 
 void
 hist_clear(histogram_t *hist) {
-  hist->used = 0;
+  int i;
+  for(i=0;i<hist->used;i++)
+    hist->bvs[i].count = 0;
+  if(hist->fast) {
+    struct histogram_fast *hfast = (struct histogram_fast *)hist;
+    for(i=0;i<256;i++)
```
Comment (rileyberton) [inline src/circllhist.c]: should have braces

File: src/circllhist.c (hunk 1 above)
Comment (rileyberton) [inline src/circllhist.c]: nevermind.. I understand

Comment (postwait) [inline src/circllhist.c]: array of pointers.  two-level cache.  2k + 512b per populated level 2.  As opposed to a single slab of 128k

**Comment by rileyberton** [comment]
Looks good to me

**Pull request merged**